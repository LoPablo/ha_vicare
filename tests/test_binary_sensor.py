"""Test ViCare binary sensors."""

from unittest.mock import MagicMock

import pytest
from syrupy.assertion import SnapshotAssertion

from homeassistant.core import HomeAssistant


@pytest.mark.freeze_time("2022-04-19 07:53:05")
@pytest.mark.parametrize(
    "entity_id",
    [
        "sensor.vicare_burner_active",
        "sensor.vicare_circulation_pump_active",
        "sensor.vicare_dhw_charging_active",
        "sensor.vicare_dhw_circulation_pump_active",
        "sensor.vicare_dhw_pump_active",
        "sensor.vicare_frost_protection_active",
    ],
)
async def test_binary_sensors(
    hass: HomeAssistant,
    mock_vicare_gas_boiler: MagicMock,
    snapshot: SnapshotAssertion,
    entity_id: str,
) -> None:
    """Test the ViCare Gas Boiler sensor."""
    state = hass.states.get(entity_id)
    assert state == snapshot
