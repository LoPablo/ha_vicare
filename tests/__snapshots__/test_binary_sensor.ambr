# serializer version: 1
# name: test_binary_sensors[sensor.vicare_burner_active]
  None
# ---
# name: test_binary_sensors[sensor.vicare_circulation_pump_active]
  None
# ---
# name: test_binary_sensors[sensor.vicare_dhw_charging_active]
  None
# ---
# name: test_binary_sensors[sensor.vicare_dhw_circulation_pump_active]
  None
# ---
# name: test_binary_sensors[sensor.vicare_dhw_pump_active]
  None
# ---
# name: test_binary_sensors[sensor.vicare_frost_protection_active]
  None
# ---
